### ISSUE TITLE
Add `plot` method for `default_priors`

### ISSUE THREAD

**Issue by GidonFrischkorn** [venpopov/bmm#194]
### Discussed in https://github.com/venpopov/bmm/discussions/191

<div type='discussions-op-text'>

<sup>Originally posted by **GidonFrischkorn** March 28, 2024</sup>
When I was thinking about the `default_priors` for the `m3`, I thought having an easy way of plotting the `default_priors` on the parameters and native scale would be nice. I was thinking about something similar to the plots you generated for the "extract info" vignette. What do you think about this?</div>

Create a `plot` method for the `default_priors` generated for `bmmodel`. Potentially see if the `plot` method can be generalized to the output class of `default_priors` and also work for `brms` models

**Comment by venpopov** [comment]
Yes, I think that package will be super useful. Here's a super basic example of how we could use it for this:

```r
library(distributional)
library(ggdist)
library(ggplot2)
library(brms)

# basic functionality with distributional explicit functions
dist <- dist_normal()
dist2 <- exp(dist).  # you can apply transformations!


# visualize both distributions
ggplot() +
  stat_slabinterval(aes(dist = dist), orientation = "horizontal")
ggplot() +
  stat_slabinterval(aes(dist = dist2), orientation = "horizontal")


# can extract the distribution from the prior and do the same
prior <- set_prior("normal(2, 1)", class = "Intercept")
dprior <- prior$prior
dprior <- paste0("distributional::dist_", dprior)

dist_prior <- eval(parse(text = dprior))

ggplot() +
  stat_slabinterval(aes(dist = dist_prior), orientation = "horizontal")
```

![image](https://github.com/venpopov/bmm/assets/12783882/ed6f7d9c-f162-4951-b0f0-13de2f2e5f9e)

**Comment by GidonFrischkorn** [comment]
Looking forward to checking this out! I saw that you also were in contact with the developers of the distributions package ☺️

**Comment by venpopov** [comment]
yeah, I had the prototype working on the weekend, but then noticed some problems related to transformations of priors to the native scale, so I've been working with them to fix them :)
